TESTS/netsocket/tcp/tcpsocket_echotest_burst.cpp

```diff
@@ -47,41 +47,35 @@ void TCPSOCKET_ECHOTEST_BURST()
     fill_tx_buffer_ascii(tcp_global::tx_buffer, BURST_SIZE);
 
     int recvd;
-    int bt_left;
     int sent;
     for (int i = 0; i < BURST_CNT; i++) {
-        bt_left = BURST_SIZE;
-        while (bt_left > 0) {
-            sent = sock.send(&(tcp_global::tx_buffer[BURST_SIZE - bt_left]), bt_left);
-            if (sent == NSAPI_ERROR_WOULD_BLOCK) {
-                if (osSignalWait(SIGNAL_SIGIO, SIGIO_TIMEOUT).status == osEventTimeout) {
-                    TEST_FAIL();
-                    goto END;
-                }
-                continue;
-            } else if (sent < 0) {
-                printf("[%02d] network error %d\n", i, sent);
+        sent = sock.send(tcp_global::tx_buffer, BURST_SIZE);
+        if (sent == NSAPI_ERROR_WOULD_BLOCK) {
+            if (osSignalWait(SIGNAL_SIGIO, SIGIO_TIMEOUT).status == osEventTimeout) {
                 TEST_FAIL();
                 goto END;
             }
-            bt_left -= sent;
+            continue;
+        } else if (sent < 0) {
+            printf("[%02d] network error %d\n", i, sent);
+            TEST_FAIL();
+            goto END;
+        } else if (sent != BURST_SIZE) {
```

Comment:
This set of conditions is inconsistent. If the call can return `WOULDBLOCK`, it can return a partial length. Either they're both acceptable (if nonblocking) or they're both not (if blocking). I'm not sure whether you're using blocking or non-blocking sockets here.

Reply:
@kjbracey-arm , thanks, this is a good point. 
The test that you commented on is a blocking one, so I removed the check for WOULD_BLOCK - we can handle it just like any other error (an fail if it shows up).
There is a non-blocking version of the echo test below this one which is allowing a WOULD_BLOCK to happen and will just retry until timeout is exceeded.
I checked that the above is true for all tcp/tls and burst/non-burst combinations.
Test are still passing.